### PR TITLE
Update Kallisto to 0.44.0.

### PR DESCRIPTION
**Pull request by cbrueffer**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/kallisto/meta.yaml [hunk 1]
```diff
@@ -7,20 +7,23 @@ package:
...
 build:
     number: 0
     string: hdf5{{CONDA_HDF5}}_{{PKG_BUILDNUM}}
 
 requirements:
     build:
-        - hdf5 {{CONDA_HDF5}}
+        - hdf5 {{ CONDA_HDF5 }}*
         - cmake
         - zlib
```
Comment (druvus) [inline recipes/kallisto/meta.yaml]: zlib line should also be updated so it is pinned

Comment (cbrueffer) [reply recipes/kallisto/meta.yaml]: Good point! Curious this didn't came up during linting.

Comment (bgruening) [reply recipes/kallisto/meta.yaml]: Me we fixed this recently ... lets try again.

Comment (bgruening) [reply recipes/kallisto/meta.yaml]: Oh wait, you don't have it as run-time dependency, so this is a static build right? If so you don't need to pin it.

**Comment by bgruening** [comment]
Seems like htslib is missing?

**Comment by cbrueffer** [comment]
Yes, needs a bit more work to find the conda htslib headers. Haven't had time to continue with this yet.

**Comment by cbrueffer** [comment]
An update was committed in https://github.com/bioconda/bioconda-recipes/commit/4ec8ca5144388aac70303e2fb7384d6c97a9c5fe so close this PR.

**Pull request closed without merging**